osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -164,23 +168,42 @@ protected sealed override void ApplyResultInternal(JudgementResult result)
             scoreResultCounts[result.Type] = scoreResultCounts.GetValueOrDefault(result.Type) + 1;
 
             if (!result.Type.IsScorable())
+            {
+                // The inverse of non-scorable (ignore) judgements may be bonus judgements.
+                if (result.Judgement.MaxResult.IsBonus())
+                    currentMaximumScoringValues.BonusScore += result.Judgement.MaxNumericResult;
```

Comment:
This sort of nested condition is making me uneasy that there's some other condition not being checked that may result in `currentMaximumScoringValues` not being fully correctly updated. I think I'd prefer that everything in `currentMaximumScoringValues` was unconditionally updated before the early return in this branch, even if it means duplicate `IsBonus()`/`AffectsCombo()`/`IsBasic()` checks...

Reply:
That's fair, it also came across that way to me also. Let me know if the latest changes match what you expect.